### PR TITLE
{toolchain}[dummy/dummy] iomkl v2017a

### PR DESCRIPTION
**Pull request by akesandgren**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/i/iompi/iompi-2017a.eb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# This is an easyconfig file for EasyBuild, see http://hpcugent.github.io/easybuild
+easyblock = "Toolchain"
+
+name = 'iompi'
+version = '2017a'
+
+homepage = 'http://software.intel.com/en-us/intel-cluster-toolkit-compiler/'
+description = """Intel C/C++ and Fortran compilers, alongside Intel MPI."""
```
Comment (boegel) [inline easybuild/easyconfigs/i/iompi/iompi-2017a.eb]: @akesandgren `s/Intel MPI/Open MPI/g`

File: easybuild/easyconfigs/i/iomkl/iomkl-2017a.eb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+version = '2017a'
+
+homepage = 'http://software.intel.com/en-us/intel-cluster-toolkit-compiler/'
+description = """Intel Cluster Toolchain Compiler Edition provides Intel C/C++ and Fortran compilers, Intel MKL & OpenMPI."""
+
+toolchain = {'name': 'dummy', 'version': 'dummy'}
+
+compver = '2017.1.132'
+gcc_maj_min = '6.3'
+gccver = '%s.0' % gcc_maj_min
```
Comment (boegel) [inline easybuild/easyconfigs/i/iomkl/iomkl-2017a.eb]: @akesandgren you can just drop `gcc_maj_min`, `gccver` and `binutilsver` not used anywhere

```python
gccsuff = '-GCC-6.3.0-2.27'
```

File: easybuild/easyconfigs/o/OpenMPI/OpenMPI-2.0.2-iccifort-2017.1.132-GCC-6.3.0-2.27.eb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+version = '2.0.2'
+
+homepage = 'http://www.open-mpi.org/'
+description = """The Open MPI Project is an open source MPI-2 implementation."""
+
+toolchain = {'name': 'iccifort', 'version': '2017.1.132-GCC-6.3.0-2.27'}
+
+sources = [SOURCELOWER_TAR_GZ]
+
+source_urls = ['http://www.open-mpi.org/software/ompi/v%(version_major_minor)s/downloads',]
```
Comment (boegel) [inline easybuild/easyconfigs/o/OpenMPI/OpenMPI-2.0.2-iccifort-2017.1.132-GCC-6.3.0-2.27.eb]: @akesandgren please include checksum like in existing OpenMPI easyconfig, and group + fix style:

```python
source_urls = ['http://www.open-mpi.org/software/ompi/v%(version_major_minor)s/downloads']
sources = [SOURCELOWER_TAR_GZ]
checksums = ['886698becc5bea8c151c0af2074b8392']
```

**Comment by boegel** [comment]
@akesandgren please also include HPL 2.2 easyconfig using this toolchain, as a test case

File: easybuild/easyconfigs/i/iompi/iompi-2017a.eb (hunk 1 above)
Comment (akesandgren) [inline easybuild/easyconfigs/i/iompi/iompi-2017a.eb]: Oops...

**Comment by boegel** [comment]
@akesandgren please also exclude `--with-threads` configure option, see #4213

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 7 out of 7 (7 easyconfigs in this PR)
node2116.delcatty.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/52449d2033f2deafaaf54f7730e137f6 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 7 out of 7 (7 easyconfigs in this PR)
node2494.golett.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/301aa58fd4fb9784763d34aa3edd6b29 for a full test report.

**Comment by boegel** [comment]
lgtm

**Comment by boegel** [comment]
Going in, thanks @akesandgren!

**Pull request merged**